### DISCUSSION TITLE
using {% highlight %}

### DISCUSSION THREAD

**Discussion by mohits** [cotes2020/jekyll-theme-chirpy#1513] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

Hi! I know that the documents say that `{% highlight %}` does not work with this theme. I wanted to get some idea of why that is the case and if there is a place I could look to try to understand if it can be made to work. I am migrating to this theme and have probably ~80 posts that use code syntax highlighting using `{% highlight %}` and want to avoid editing all the posts since the theme is fantastic other than that!

Thanks!

### Operations you have already tried

* Reading the docs

**Comment by mohits** [comment]
For example, I found that the highlight bit in the snapshot below is not included in the output when using `{% highlight %}` and if I add it back to the generated HTML, it seems to fix the issue.

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/56816/22d31e53-bcf4-4cde-96df-e4b518079c31)

But, currently, I don't know where to look to fix this since the code in Jekyll looks fine, of course.

**Comment by cotes2020** [comment]
Jekyll generates different HTML structures for `` ``` `` and `{% highlight %}`, and `_includes/refactor-content.html` changes the original HTML structure to add functionality. 

 If you are looking to adapt the `{% highlight %}` tag, consider modifying that file.

**Reply by mohits** [reply]
Thanks, I will continue to take a look and see what makes sense.

**Reply by mohits** [reply]
Thanks for the tip. I looked at it, and at least for my case (using `{% highlight %}` with Textile markup and `Rouge` for code highlighting, the rendering works perfectly when I remove this:

```html
<!--
  Fixed kramdown code highlight rendering:
  https://github.com/penibelst/jekyll-compress-html/issues/101
  https://github.com/penibelst/jekyll-compress-html/issues/71#issuecomment-188144901
-->

{% if _content contains '<pre class="highlight">' %}
  {% assign _content = _content
    | replace: '<div class="highlight"><pre class="highlight"><code', '<div class="highlight"><code'
    | replace: '</code></pre></div>', '</code></div>'
  %}
{% endif %}
```

Of course, there is a comment there that this probably breaks kramdown and so, this has to be carefully looked at :(
